### DISCUSSION TITLE
program tcp_close: map events: map create without BTF: invalid argument

### DISCUSSION THREAD

**Discussion by zhuanshenlikai** [cilium/ebpf#802] [Q&A] [answered]
I try the  tcprtt from the examples, but return these errors. I use linux version of ubuntu(5.4.0-121-generic).

![image](https://user-images.githubusercontent.com/12570095/192182388-6ca6f21a-780d-4c72-9e55-7ac817a1deb9.png)

Is there anything wrong?

**Comment by brycekahle** [comment, marked as answer]
5.4 is not new enough for that example. The helper `bpf_skc_to_tcp_sock` is not available until 5.9 and the `fentry` program type was added in 5.5.

**Reply by brycekahle** [reply]
Oh, and ring buffer was added in 5.8.

**Reply by zhuanshenlikai** [reply]
thanks
